python/metatensor-operations/metatensor/operations/sort.py

```diff
@@ -0,0 +1,237 @@
+from typing import List, Union
+
+from . import _dispatch
+from ._classes import Labels, TensorBlock, TensorMap
+
+
+def _sort_single_block(
+    block: TensorBlock,
+    axes: List[str],
+    descending: bool,
+) -> TensorBlock:
+    """
+    Sorts a single TensorBlock without the user input checking and sorting of gradients
+    """
+
+    samples_names = block.samples.names
+    samples_values = block.samples.values
+
+    components_names: List[List[str]] = []
+    components_values = []
+    for component in block.components:
+        components_names.append(component.names)
+        components_values.append(component.values)
+
+    properties_names = block.properties.names
+    properties_values = block.properties.values
+
+    values = block.values
+    if "samples" in axes:
+        sorted_idx = _dispatch.argsort_metadata_values(
+            samples_values, reverse=descending
+        )
+        samples_values = samples_values[sorted_idx]
+        values = values[sorted_idx]
+    if "components" in axes:
+        for i, _ in enumerate(block.components):
+            sorted_idx = _dispatch.argsort_metadata_values(
+                components_values[i], reverse=descending
+            )
+            components_values[i] = components_values[i][sorted_idx]
+            values = _dispatch.take(values, sorted_idx, axis=i + 1)
+    if "properties" in axes:
+        sorted_idx = _dispatch.argsort_metadata_values(
+            properties_values, reverse=descending
+        )
+        properties_values = properties_values[sorted_idx]
+        values = _dispatch.take(values, sorted_idx, axis=-1)
+
+    samples_labels = Labels(names=samples_names, values=samples_values)
+    properties_labels = Labels(names=properties_names, values=properties_values)
+    components_labels = [
+        Labels(names=components_names[i], values=components_values[i])
+        for i in range(len(components_names))
+    ]
+
+    return TensorBlock(
+        values=values,
+        samples=samples_labels,
+        components=components_labels,
+        properties=properties_labels,
+    )
+
+
+def sort_block(
+    block: TensorBlock,
+    axes: Union[str, List[str]] = "all",
+    descending: bool = False,
+) -> TensorBlock:
+    """
+    Rearanges the values of an block according to the order given by the sorted metadata
+    of the given axes.
+
+    This function creates copies of the metadata on the CPU to sort the metadata.  For
+    that each row have to be interpreted as tuple which is not supported by torch.
```

Comment:
```suggestion
    This function creates copies of the metadata on the CPU to sort the metadata.
```

users of the function should not have to care about this (but this could be a comment inside the function, or we can implement sort in metatensor-core)